### PR TITLE
Update the product model custom option methods PHPdoc

### PR DESCRIPTION
**Pull request by hws47a**
### Description (*)
Improve custom option related method phpdoc type hints.
Simplify some custom option related methods.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by lenaorobei — COMMENTED** [review]
Hi @hws47a!
Thanks for this improvement. Could you please address static tests failures?
https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/26697/8a97a81b52dae423c72a9de5d24bc3e5/Statics/allure-report-b2b/index.html#suites/51d3ab4d5d77a0bc6a7ea344e9e7e6be/b8240da05cffedb7/

Thanks!

**Comment by hws47a** [comment]
Hi @lenaorobei, I've fixed PHPStan errors although they don't have anything in common with my change.
There is another PHPCompatibility issue with `__toArray` method, but it's a public one so I'm not sure if it's allowed to remove it.

**Comment by lenaorobei** [comment]
Let's add ignore for this case `//phpcs:ignore PHPCompatibility.FunctionNameRestrictions.ReservedFunctionNames`

**Comment by hws47a** [comment]
@lenaorobei done

**Comment by lenaorobei** [comment]
@hws47a could you please check failed integration test?

File: app/code/Magento/Catalog/Model/Product.php [hunk 1]
```diff
@@ -2340,7 +2334,8 @@ public function isDisabled()
     public function getImage()
     {
         $this->getTypeInstance()->setImageFromChildProduct($this);
-        return parent::getImage();
+
+        return (string)$this->getData('image');
```
Comment (hws47a) [inline app/code/Magento/Catalog/Model/Product.php]:
```suggestion
        return $this->getData('image');
```

**Comment by hws47a** [comment]
@lenaorobei done

**Comment by hws47a** [comment]
@lenaorobei there is a conflict caused by another fix of the same Static test errors.
I think while merging you can choose which way of fixing you prefer.

**Comment by sivaschenko** [comment]
SVC failure is false positive. Functional Tests EE failure looks random

**Review by sivaschenko — DISMISSED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-6921](https://jira.corp.magento.com/browse/ENGCOM-6921) has been created to process this Pull Request
:eight_spoked_asterisk: @sivaschenko, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by sivaschenko** [comment]
Product custom options functionality is covered with various MFTF tests, i.e. 
`SimpleProductTwoCustomOptionsTest`,
`AdminRemoveCustomOptionsFromProductTest`, `SaveProductWithCustomOptionsAdditionalWebsiteTest`, `SaveProductWithCustomOptionsAdditionalWebsiteTest`, etc.

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed

**Comment by slavvka** [comment]
@sivaschenko Why is the SVC failure false positive? There are changes in the contract of the API class.

**Comment by sivaschenko** [comment]
@slavvka because the pull request does not contain any backward incompatible changes.

Approval request for false positive SVC failure: https://jira.corp.magento.com/browse/MC-31715

File: app/code/Magento/Catalog/Model/Product.php [hunk 2]
```diff
@@ -2084,14 +2085,11 @@ public function getCustomOptions()
      * Get product custom option info
      *
      * @param   string $code
-     * @return  array
+     * @return  OptionInterface|null
```
Comment (naydav) [inline app/code/Magento/Catalog/Model/Product.php]: Theoretically, this class can be used in WebAPI requests.

The current implementation of WebAPI "request/response" parser has a limitation:

Valid object types include a fully qualified class name or a fully qualified interface name.
https://devdocs.magento.com/guides/v2.3/extension-dev-guide/service-contracts/service-to-web-service.html

It's OK to change dockblock but please replace short namespace on "fully qualified name" in methods dockblocks
Thanks

**Review by naydav — DISMISSED** [review]
Need to change dockblocks. See my comment above. Thanks.

**Review by slavvka — DISMISSED** [review]
Please fix the code according to @naydav comment

**Review by sivaschenko — DISMISSED** [review]
@slavvka @naydav review appears to be not actual

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-6921](https://jira.corp.magento.com/browse/ENGCOM-6921) has been created to process this Pull Request

**Comment by sivaschenko** [comment]
SVC failure is approved: https://jira.corp.magento.com/browse/MC-31715

**Review by sivaschenko — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-6921](https://jira.corp.magento.com/browse/ENGCOM-6921) has been created to process this Pull Request

**Pull request merged**